File: frontend/src/types/activity.ts
/**
 * Activity Feed & Audit Log Contracts matching backend activity schemas.
 */

export interface ActivityResponse {
  id: string;
  user_id: string;
  action: string;
  resource_type: string;
  resource_id: string;
  metadata_json?: string | null;
  ip_address?: string | null;
  created_at: string;
}

export interface ActivityListResponse {
  items: ActivityResponse[];
  total: number;
  page: number;
  page_size: number;
}

File: backend/app/schemas/activity.py
"""Audit Trail Activity & Security Audit Pydantic v2 Schemas."""

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class ActivityResponse(BaseModel):
    """Single user activity / security audit event payload."""

    id: uuid.UUID = Field(..., description="Unique event identifier")
    user_id: uuid.UUID = Field(..., description="Actor user identifier")
    action: str = Field(..., description="Action name or event type")
    resource_type: str = Field(
        ..., description="Target resource domain type (FILE, FOLDER, USER, SYSTEM)"
    )
    resource_id: uuid.UUID = Field(..., description="Target resource identifier")
    metadata_json: str | None = Field(default=None, description="Structured JSON event metadata")
    ip_address: str | None = Field(default=None, description="Actor IP address if logged")
    created_at: datetime = Field(..., description="Timestamp when event occurred")

    model_config = ConfigDict(from_attributes=True)


class ActivityListResponse(BaseModel):
    """Paginated list of activity or security audit records."""

    items: list[ActivityResponse] = Field(default_factory=list, description="Activity records")
    total: int = Field(..., description="Total matching activity records")
    page: int = Field(default=1, description="Current page index")
    page_size: int = Field(default=50, description="Items per page")

    model_config = ConfigDict(from_attributes=True)
